input_definition.go

```diff
@@ -111,14 +115,27 @@ func (i *InputDefinition) LoadDefinition(pb *internal.InputDefinition) error {
 			})
 		}
 
+		if field.PrimaryKey {
+			// Deprecating column labels per #810.
+			// So, setting the default column label here.
+			fieldName = DefaultColumnLabel
+			primaryKeyGiven = true
+		}
+
 		inputField := InputDefinitionField{
-			Name:       field.Name,
+			Name:       fieldName,
 			PrimaryKey: field.PrimaryKey,
 			Actions:    actions,
 		}
 		i.fields = append(i.fields, inputField)
 	}
 
+	if len(pb.Fields) > 0 && !primaryKeyGiven {
+		// primary field is required if there are other fields.
+		// add it if it doesn't exist.
+		i.fields = append(i.fields, InputDefDefaultPrimaryKeyField())
```

Comment:
It seems confusing to add a field to the input definition. Why not just return an error if a primary key is not set?

Reply:
It's convenient to leave out the primary key in the definition, since its name would always be the default column label (currently `columnID`). If we don't add the primary key field auto, the user has to add this to the following to `fields` for any input def:
```json
{
  "name": "columnID",
  "primaryKey": true
}
```